### DISCUSSION TITLE
Emmitting "side effect" deletion events

### DISCUSSION THREAD

**Discussion by Jonesus** [directus/directus#4841] [Feature Requests]
Currently (AFAIK) deletion events are only emitted if an item is the direct target of the action (for example deleting an item emits an deletion event for that item), but there can also be "side effects" when deleting some items, for example deleting a collection also deletes its fields, relations and more: 

https://github.com/directus/directus/blob/ecd5526b55aaab3b04e20f9e21285df3600b02ca/api/src/services/collections.ts#L290-L296

Should we also emit events for each of these "side effects"? IMO an event should be fired whenever something happens to an item that a hook is listening to, for example if I have a hook that is listening to deletions of any fields, I would want it to also fire when fields get deleted because their collection got deleted. I could work on a PR for this functionality if you agree with how I envision the functionality :smiley_cat: 

One problem with emitting deletes in a situation like this is that do we want to block the deletion of a collection if we have, for example,  a hook listening to `activity.delete.before`, which would be a side-effect of the collection's deletion?

Thoughts?

**Comment by sabinehoi** [comment]
I ran into the same issue! Assumed my hook would _always_ work but found a bug that was caused by the hook not being triggered on cacade deletes

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
